File: Algorithms/Sorting/bubble_sort.rs
fn bubble_sort(seq: &mut [u8; 5]) {
    for i in 0..seq.len() - 1 {
        for j in 0..seq.len() - 1 {
            if seq[j] > seq[j + 1] {
                seq.swap(j, j + 1);
            }
        }
    }
}

fn main() {
    let mut seq: [u8; 5] = [3, 5, 2, 4, 1];

    bubble_sort(&mut seq);

    for i in &seq {
        print!("{} ", i);
    }
    print!("\n");
}

File: Algorithms/Sorting/bubble_sort.go
package main

import "fmt"

func bubble_sort(seq []int) {
  for i := 0; i < len(seq) - 1; i++ {
    for j := 0; j < len(seq) - 1; j++ {
      if seq[j] > seq[j + 1] {
        seq[j], seq[j + 1] = seq[j + 1], seq[j]
      }
    }
  }
}

func main() {
  seq := []int{3, 5, 2, 4, 1}

  bubble_sort(seq)
  fmt.Println(seq)
}
